File: frontend/src/dtos/createMessageDto.ts
import type { RabbitMQNewMessageMetaDto } from './rabbitMQNewMessageMetaDto'

export interface CreateMessageDto {
  brokerId: string
  queueId: string
  body: any
  bodyEncoding: 'json' | 'base64'
  rabbitmqMetadata?: RabbitMQNewMessageMetaDto
}

File: backend/Resqueue/Resqueue/Dtos/CreateMessageDto.cs
using System.Text.Json.Nodes;

namespace Resqueue.Dtos;

public class NewMessageDto
{
    public string BrokerId { get; set; }
    public JsonNode Body { get; set; }

    /// <summary>
    ///  "json" or "base64"
    /// </summary>
    public string BodyEncoding { get; set; }

    public RabbitmqNewMessageMetadataDto? RabbitmqMetadata { get; set; }
}